coap-gateway/service/client.go

```diff
@@ -334,19 +334,28 @@ func (client *Client) cancelResourceSubscriptions(wantWait bool) {
 	}
 }
 
-func (client *Client) cancelDeviceSubscriptions(wantWait bool) {
-	deviceSubscriptions := client.deviceSubscriptions.PullOutAll()
-	for _, v := range deviceSubscriptions {
-		o, ok := grpcClient.ToDeviceSubscription(v.(*atomic.Value).Load(), true)
-		if !ok {
-			continue
-		}
-		wait, err := o.Cancel()
-		if err != nil {
-			log.Errorf("cannot cancel device subscription: %v", err)
-		} else if wantWait {
-			wait()
-		}
+func (client *Client) pullOutDeviceSubscription() func(ctx context.Context) error {
```

Comment:
What excactly is pullOut supposed to do? Pull out somehow doesn't describe it clearly.